tracer/src/Datadog.Trace/ClrProfiler/Integrations/AspNet/AspNetWebApi2Integration.cs

```diff
@@ -145,6 +148,108 @@ public static class AspNetWebApi2Integration
                 cancellationToken);
         }
 
+        /// <summary>
+        /// Calls the underlying HandleAsync and traces the request.
+        /// </summary>
+        /// <param name="handler">The exception handler</param>
+        /// <param name="context">The exception context for the call</param>
+        /// <param name="boxedCancellationToken">The cancellation token</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <param name="moduleVersionPtr">A pointer to the module version GUID.</param>
+        /// <returns>A task with the result</returns>
+        [InterceptMethod(
+            TargetAssembly = SystemWebHttpAssemblyName,
+            TargetType = ExceptionHandlerExtensionsTypeName,
+            TargetSignatureTypes = new[] { ClrNames.HttpResponseMessageTask, IExceptionHandlerTypeName, ExecutionContextTypeName, ClrNames.CancellationToken },
+            TargetMinimumVersion = Major5Minor1,
+            TargetMaximumVersion = Major5MinorX)]
+        public static object HandleAsync(
+            object handler,
+            object context,
+            object boxedCancellationToken,
+            int opCode,
+            int mdToken,
+            long moduleVersionPtr)
+        {
+            if (handler == null) { throw new ArgumentNullException(nameof(handler)); }
+
+            var cancellationToken = (CancellationToken)boxedCancellationToken;
+            var handlerType = handler.GetInstrumentedInterface(IExceptionHandlerTypeName);
+
+            Type taskResultType;
+
+            try
+            {
```

Comment:
This try/catch block is largely copied from the `ExecuteAsync` method above, which uses the runtime types to find the loaded `System.Net.Http` assembly and find the `System.Net.Http.HttpRequestMessage` type so we can return a `Task<T>` with the appropriate generic type